Migrate x86 trunk build/test to macos12

albanD opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #77662
* #77647
* __->__ #77645
* #77644

This will enable MPS building but will NOT test mps
as the runner do not have AMD gpus

Note that we disable nnpack for now due to https://github.com/pytorch/pytorch/issues/76094

albanD committed:
Migrate x86 trunk build/test to macos12

This will enable MPS building but will NOT test mps
as the runner do not have AMD gpus

[ghstack-poisoned]

malfet commented on .github/workflows/_mac-build.yml:
```diff
@@ -69,6 +69,7 @@ jobs:
           auto-update-conda: true
           python-version: 3.8
           activate-environment: build
+          miniconda-version: latest
```
Hmm, can you please specify the exact version here? So that we would not accidentally regress when miniconda updates

albanD replied:
Updated

seemethere reviewed (approved):
Agree with Nikita's comment about pinning miniconda version

albanD committed:
Update on "Migrate x86 trunk build/test to macos12"


This will enable MPS building but will NOT test mps
as the runner do not have AMD gpus

Note that we disable nnpack for now due to https://github.com/pytorch/pytorch/issues/76094

[ghstack-poisoned]

albanD commented:
Added pinning for both conda and xcode versions

albanD commented:
@pytorchbot merge this please
